Jenkinsfile.update-repo-list

```diff
@@ -14,8 +15,16 @@ elifePipeline {
             gitCommitRef = params.gitCommitRef
         }
 
+        stage 'Checking if the Repo List file Changed', {
+            isRepoListChanged = sh(script: "make check-repo-list-file-change", returnStdout: true).trim()
```

Comment:
One minor disadvantage with doing this using the Jenkinsfile is that it's difficult to test locally. Although Groovy definitely has a better programming experience than bash or make..

In this case I am not fully clued up on the behaviour of [sh](https://www.jenkins.io/doc/pipeline/steps/workflow-durable-task-step/#sh-shell-script) with exit codes. I believe it will fail, unless you use `returnStatus`. Also be mindful that `make` might also add additional output, such as the command being run. Unless you suppress it, e.g. via adding `--quiet` (I think) or prefixing the command with `@`.

Reply:
Yes you are right it is hard to test in local, but in `Makefile` using `if` statement was a bit confusing so I decided to do it in `Jenkins` file. If you say that check would be better in `Makefile` I can continue to try :D 

> I believe it will fail, unless you use `returnStatus`. 

Sorry, I did not get to point to should be used `returnStatus` here. 

> Also be mindful that make might also _add additional output_ 

Yes thank you for remind it! I will change the code to make it quite.

Reply:
> Yes you are right it is hard to test in local, but in `Makefile` using `if` statement was a bit confusing so I decided to do it in `Jenkins` file. If you say that check would be better in `Makefile` I can continue to try :D

I added an example further down. I am not sure whether it's better. The main reason is really so that you could run it locally.

> 
> > I believe it will fail, unless you use `returnStatus`.
> 
> Sorry, I did not get to point to should be used `returnStatus` here. Can you explain to me to fix it?
> I get the idea from https://stackoverflow.com/a/51216603/12570042

I believe the `git rev-list HEAD --count` wouldn't usually return a non-zero exit status (e.g. `1`), like `git diff` does.
Like in a makefile, I believe the default behaviour of the `sh` in a Jenkinsfile is to fail, if the executed command returned a non-zero exit status. By adding `returnStatus` you can capture the exit code and it won't fail. Of course you could also make the command always succeed. There seem to be a few moving parts that need to work together.